lib/health_check.rb

```diff
@@ -0,0 +1,97 @@
+class HealthCheck
+  include ChefResource
+
+  OK = 'ok'.freeze
+  NOT_OK = 'not ok'.freeze
+  REACHABLE = 'reachable'.freeze
+  TIMEOUT = 'timeout'.freeze
+  UNREACHABLE = 'unreachable'.freeze
+  ERRORING = 'erroring'.freeze
+  AUTHERROR = 'authentication error'.freeze
+
+  attr_reader :status, :erchef, :postgres
+
+  def initialize
+    @status = OK
+    @erchef = { status: REACHABLE }
+    @postgres = { status: REACHABLE }
+  end
+
+  #
+  # Do a general health check.
+  #
+  def check
+    erchef_health
+    erchef_authentication
+    postgres_health
+    overall
+  end
+
+  #
+  # Shortcut to see if all is well
+  #
+  def ok?
+    @status == OK
+  end
+
+  private
+
+  #
+  # Try to ping erchef as an indicator of general health
+  #
+  def erchef_health
+    erchef_health_metric do
+      chef.get_rest '_status'
+    end
+  end
+
+  #
+  # Try to authenticate with erchef to prove that oc-id is actually functioning
+  #
+  # TODO: in order for this to work, we need a test user, with
+  # username/password stored in the vault and accessible in Settings
+  #
+  def erchef_authentication
```

Comment:
From my point of view, this endpoint works fine without this, but having this would make the endpoint a lot more useful.  It doesn't feel like that big of a deal to setup, but maybe I'm missing something.  @smith - do you have any opinions on this?